src/components/Header/AuthenticatedHeaderOption.js

```diff
@@ -0,0 +1,464 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Link, NavLink } from "react-router-dom";
+import { PATHS } from "../../constant";
+import axios from "axios";
+import { METHODS } from "../../services/api";
+import { hasOneFrom } from "../../common/utils";
+import { actions as userActions } from "../User/redux/action";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import ExpandLessIcon from "@mui/icons-material/ExpandLess";
+import useStyles from "./styles";
+import { DropDown, MobileDropDown } from "./DropDown";
+import { breakpoints } from "../../theme/constant";
+import useMediaQuery from "@mui/material/useMediaQuery";
+
+import {
+  Box,
+  IconButton,
+  Typography,
+  Menu,
+  Avatar,
+  MenuItem,
+} from "@mui/material";
+
+const SwitchView = ({
+  role,
+  setSwitchView,
+  handleCloseSwitchView,
+  switchView,
+}) => {
+  const classes = useStyles();
+  if (role == "admin") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.PARTNERS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  if (role == "volunteer") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.CLASS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  if (role == "partner") {
+    return (
+      <MenuItem
+        onClick={() => {
+          setSwitchView(role);
+          handleCloseSwitchView();
+        }}
+        sx={{ margin: "0px 10px 0px 10px" }}
+        className={switchView === role && classes.bgColor}
+      >
+        <NavLink to={PATHS.PARTNERS} className={classes.link}>
+          {role}
+        </NavLink>
+      </MenuItem>
+    );
+  }
+  return "";
```

Comment:
Using the suggestion about the constant with the landing pages for each role (https://github.com/navgurukul/bhanwari-devi/pull/290#discussion_r846607757):
```suggestion
  const roleLandingPage = rolesLandingPages[role];
  return roleLandingPage ?
    return (
      <MenuItem
        onClick={() => {
          setSwitchView(role);
          handleCloseSwitchView();
        }}
        sx={{ margin: "0px 10px 0px 10px" }}
        className={switchView === role && classes.bgColor}
      >
        <NavLink to={roleLandingPage} className={classes.link}>
          {role}
        </NavLink>
      </MenuItem>
    ) : "";